kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java

```diff
@@ -0,0 +1,307 @@
+        if (pollService != null) {
+            long delay = Long.getLong(KAFKA_EXTENSION_PREFIX + "poll.interval", 1L);
+            pollService.scheduleWithFixedDelay(this, delay, delay, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void createContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        // will be done by listener
+    }
+
+    @Override
+    public boolean isUpdateContainerAllowed(String id,
+                                            KieContainerInstance kieContainerInstance,
+                                            Map<String, Object> parameters) {
+        return false;
+    }
+
+    @Override
+    public void prepareContainerUpdate(String id,
+                                       KieContainerInstance kieContainerInstance,
+                                       Map<String, Object> parameters) {
+        // will be done by listener
+    }
+
+    @Override
+    public void updateContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        // will be done by listener
+    }
+
+    @Override
+    public void disposeContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+        // will be done by listener
+    }
+
+    @Override
+    public List<Object> getAppComponents(SupportedTransports type) {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public <T> T getAppComponents(Class<T> serviceType) {
+        return null;
+    }
+
+    @Override
+    public String getImplementedCapability() {
+        return KieServerConstants.CAPABILITY_BPM_KAFKA;
+    }
+
+    @Override
+    public List<Object> getServices() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public String getExtensionName() {
+        return EXTENSION_NAME;
+    }
+
+    @Override
+    public Integer getStartOrder() {
+        return 20;
+    }
+
+    @Override
+    public void run() {
+        if (consumer != null && !topic2Signal.isEmpty()) {
+            ConsumerRecords<String, byte[]> events;
+            synchronized (consumer) {
+                events = consumer.subscription().isEmpty() ? ConsumerRecords.empty() : consumer.poll(Duration.ZERO);
+            }
+            for (ConsumerRecord<String, byte[]> event : events) {
+                notifyService.submit(() -> processEvent(event));
+            }
+        }
+    }
+
+    private void processEvent(ConsumerRecord<String, byte[]> event) {
+        SignalInfo signalInfo = topic2Signal.get(event.topic());
+        if (signalInfo != null) {
+            try {
+                String type = signalInfo.getSignalDesc().getName();
+                if (signalInfo.getSignalDesc().getSignalType() == SignalType.MESSAGE) {
+                    type = "Message-" + type;
+                }
+                processService.signalEvent(signalInfo.getDeploymentId(), type, CloudEvent.read(event.value(),
```

Comment:
debug levels regarding what signal you received and where are you sending it.
trace would be the message content.

Reply:
done